File: L237-delete_node_in_a_linked_list.py
# https://leetcode.com/problems/delete-node-in-a-linked-list
class Solution:
    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        while (node.next):
            node.val = node.next.val
            prev = node
            node = node.next
        prev.next = None

File: L237-DeleteNodeInALinkedList.java
// https://leetcode.com/problems/delete-node-in-a-linked-list
class Solution {
    public void deleteNode(ListNode node) {
        ListNode prev = new ListNode();
        while (node.next != null) {
            node.val = node.next.val;
            prev = node;
            node = node.next;
        }
        prev.next = null;
    }
}
